lib/worker.js

```diff
@@ -150,6 +155,23 @@ async function updateDependency(upgrade) {
       newVersion);
   }
 
+  function getYarnLockFileName(packageFileName) {
+    const packagePath = packageFileName.split('/');
+    // Remove 'package.json'
+    const basePath = packagePath.slice(0, -1);
+    basePath.push('yarn.lock');
+    return basePath.join('/');
+  }
+
+  function generateLockFile(newPackageJson) {
+    logger.debug('Generating new yarn.lock file');
+    rimraf.sync('/tmp/renovate-yarn');
+    mkdirp.sync('/tmp/renovate-yarn');
+    fs.writeFileSync('/tmp/renovate-yarn/package.json', newPackageJson);
+    cp.spawnSync('yarn', ['install'], { cwd: '/tmp/renovate-yarn' });
```

Comment:
Do you just mean copying the `.npmrc` file out of the *repository* - like we do with `package.json` - or something else? I assume that's what you mean by "`.npmrc` file from the repo" but didn't think there would be any auth tokens checked into source?